src/main/java/teammates/logic/core/FeedbackResponsesLogic.java

```diff
@@ -201,7 +201,7 @@ public boolean isNameVisibleToUser(
         // Early return if user is giver
         if (question.giverType == FeedbackParticipantType.TEAMS) {
             // if response is given by team, then anyone in the team can see the response
-            if (roster.isStudentsInSameTeam(response.giver, userEmail)) {
```

Comment:
The old code works for legacy data for team response.

Does it work for the new data (where giver is the team name)? It looks like a `no` to me.

Reply:
Okay I think I get it now haha, have pushed new code to check for both current and legacy cases